doc/audit/security_auditing_HLD.md

```diff
@@ -0,0 +1,366 @@
+  ```
+  admin@sonic:~$ show audit
+  -a always,exit -F arch=b32 -S exit,execve,exit_group -F auid>1000 -F auid!=-1 -F key=tacplus
+  -a always,exit -F arch=b64 -S execve,exit,exit_group -F auid>1000 -F auid!=-1 -F key=tacplus
+  -a never,exit -F path=/usr/bin/docker -F perm=x -F key=process_audit
+  -a never,exit -F path=/usr/bin/dockerd -F perm=x -F key=process_audit
+  -a never,exit -F path=/usr/bin/containerd -F perm=x -F key=process_audit
+  -a never,exit -F path=/usr/bin/runc -F perm=x -F key=process_audit
+  -a never,exit -F path=/usr/bin/python* -F perm=x -F key=process_audit
+  -a always,exit -F arch=b64 -S setuid,setresuid,setreuid,setfsuid,setgid,setresgid,setregid,setfsgid -F key=user_group_management
+  -a always,exit -F arch=b32 -S setuid,setresuid,setreuid,setfsuid,setgid,setresgid,setregid,setfsgid -F key=user_group_management
+  -a exit,always -F arch=b64 -S unlink -S unlinkat -F key=file_deletion
+  -a exit,always -F arch=b32 -S unlink -S unlinkat -F key=file_deletion
+  -a always,exclude -F msgtype=CRED_ACQ
+  -a always,exclude -F msgtype=CRED_DISP
+  -a always,exclude -F msgtype=CRED_REFR
+  -a always,exclude -F msgtype=CWD
+  -a always,exclude -F msgtype=LOGIN
+  -a always,exclude -F msgtype=PATH
+  -a always,exclude -F msgtype=PROCTITLE
+  -a always,exclude -F msgtype=SERVICE_START
+  -a always,exclude -F msgtype=SERVICE_STOP
+  -a always,exclude -F msgtype=USER_ACCT
+  -a always,exclude -F msgtype=USER_AUTH
+  -a always,exclude -F msgtype=USER_CMD
+  -a always,exclude -F msgtype=USER_END
+  -a always,exclude -F msgtype=USER_START
+  ```
+  If security auditing is disabled, `show audit` will only show tacplus accounting rule.
+  ```
+  admin@sonic:~$ show audit
+  -a always,exit -F arch=b32 -S exit,execve,exit_group -F auid>1000 -F auid!=-1 -F key=tacplus
+  -a always,exit -F arch=b64 -S execve,exit,exit_group -F auid>1000 -F auid!=-1 -F key=tacplus
+  -a always,exclude -F msgtype=CRED_ACQ
+  -a always,exclude -F msgtype=CRED_DISP
+  -a always,exclude -F msgtype=CRED_REFR
+  -a always,exclude -F msgtype=CWD
+  -a always,exclude -F msgtype=LOGIN
+  -a always,exclude -F msgtype=PATH
+  -a always,exclude -F msgtype=PROCTITLE
+  -a always,exclude -F msgtype=SERVICE_START
+  -a always,exclude -F msgtype=SERVICE_STOP
+  -a always,exclude -F msgtype=USER_ACCT
+  -a always,exclude -F msgtype=USER_AUTH
+  -a always,exclude -F msgtype=USER_CMD
+  -a always,exclude -F msgtype=USER_END
+  -a always,exclude -F msgtype=USER_START
+  ```
+
+**config command**
+- `config audit enable` -  enables all **security** audit rules (`security-auditing.rules`).
+  ```
+  admin@sonic:~$ config audit enable
+  Security auditing is enabled.
+  ```
+- `config audit disable` - removes or disables all **security** audit rules.
+  ```
+  admin@sonic:~$ config audit disable
+  Security auditing is disabled.
+  ```
+
+#### 3.3.4 Logrotate
+The following settings in the `/etc/logrotate.d/audit` file set up log rotation for audit logs:
+```
+{
```

Comment:
Do you need to specify the log filename?

Reply:
Updated